### DISCUSSION TITLE
Identical Seed produces different image?

### DISCUSSION THREAD

**Discussion by Omenizer** [adieyal/sd-dynamic-prompts#266] [Q&A]
When I run a batch and then select one image from it for Hires fix, dynamic prompts will create a different combination than within the batch.

Am I doing something wrong?

**Comment by adieyal** [comment]
Have you tried to check the "Fixed seed" checkbox under advanced settings?

**Comment by Omenizer** [comment]
Yes, but then it's using the same seed and producing different results for each run? 

I want to have the same results for the same seed, but I'm not getting them after running a batch.

What I want to do is run a large lowres batch, pick the best random creation and then upscale the best seed with Hires Fix.

**Comment by adieyal** [comment]
It's working for me.  To test

1. Prompt: "A red rose"
2. Seed: 1000
3. Batch count: 2
4. Batch size: 2
5. Generate

then generate a second time and compare images within a batch, they're different, but between batches and they're the same.  Are you getting a different result?

**Reply by Omenizer** [reply]
Sorry, I wasn't being clear - I was talking about the "dynamic prompts" extension, not vanilla Auto1111.

Prompting: {red|blue|granite} {rose|dog|battleship} with seed 1000 will give me a different combo in a batch than when I rerun a single prompt later on.

**Comment by adieyal** [comment]
Perhaps I wasn't being clear either :) I was talking about dynamic prompts (even if there aren't any dynamic components in your prompt, they are still being processed by DP). 

The way DP works is that the prompt is incremented by one for every image in your batch. If you set your seed to 1, then batch count to 4 and batch size to 4, your last image will have a seed of 16. You can confirm this by looking at the metadata embedded in your png (if you have it enabled).  

Here is what I get using your prompt with a seed of 1 and the standard 1.5 model:

![grid-0595-1-blue rose](https://user-images.githubusercontent.com/420786/220866493-0553f243-ccac-4088-aae3-9b8af53a357e.png)


Here is the metadata for the last image:

![Screenshot from 2023-02-23 11-20-33](https://user-images.githubusercontent.com/420786/220866580-60474b15-a5d2-453d-bd0a-50bbe3c674c8.png)

I can now use those same parameters to re-create that image.

Are you seeing something different?

**Reply by Omenizer** [reply]
Yes, I just tried it several times. If I pick any seed except for the 1st one, Dynamic Prompts will use a different combination of prompt tokens than when the seed is used in the batch.

I was under the impression that DP will use the seed to randomize the prompt items, since there's an "unlink from seed" option.

Can you recreate the blue rose by just using the same seed? I can't

**Comment by adieyal** [comment]
The default behaviour is for the seed to control both the prompts as well as Auto1111's image generation. If you use the same seed and press generate repeatedly, you will get exactly the same images.

When you check the unlink seed, it will generate random prompts and only use the seed for image generation.

To be clear, if you configure these settings, you are saying that you don't get a blue rose?

Prompt: blue rose
Steps: 20
Sampler: Euler a
CFG Scale: 7
Seed: 16
Sie: 512x512
Model hash: cc6cb27103

**Comment by Omenizer** [comment]
1) set seed to -1, generate batch
2) use seed from batch without changing prompt -> different result. 

Are you getting the same result?

**Reply by adieyal** [reply]
Say you set seed to 999, then generate four images using this prompt `A {red|green|blue} rose`. The images will be created with the following seeds:

Image 1: 999
Image 2: 1000
Image 3: 1001
Image 4: 1002

Now If I set the seed to 1001 and generate one image, I will get image 3. 

Are you making sure to choose seed 1001 and not 999?

**Comment by goodlux** [comment]
I'm pretty new to SD, but I read somewhere that if you have --xformers option enabled, the results are non-deterministic, so you might not get the exact same image from the same seed. If you've tried everything else and nothing works, maybe try disabling --xformers in the  webui-user.sh settings and restart ./webui.sh.  Maybe this will help!

**Reply by goodlux** [reply]
more about that in this very detailed thread https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/2958?sort=top

**Comment by scrumpyman** [comment]
Just to clarify that this is indeed an issue: https://github.com/adieyal/sd-dynamic-prompts/issues/320

**Comment by PachOOL** [comment]
The solution is to disable option: "Shuffle wildcards before use for more random outputs"
